cypress/e2e/GUI/pagination_and_elements_pag5.cy.js

```diff
@@ -0,0 +1,36 @@
+/// <reference types="Cypress" />
+describe('Paginação', () => {
+    beforeEach(() => {
+      for (let page = 1; page <= 5; page++) {
+        cy.intercept(
+          'GET',
+          `/customers?page=${page}&limit=10&size=All`,
+          { fixture: `page${page}&limit10All.json` }
+        ).as(`pag${page}limit10all`);
+      }
+      cy.visit('/') 
+      cy.wait('@pag1limit10all');
+    });
+    context('Quando valido que estou na página 1', () => {
+        it('Então deve conter página 1 de 5', () => {
+            cy.get('[data-testid="pagination"] > span').then(($confirmElement) => {
+                if($confirmElement.length > 0) {
+                    cy.get('[data-testid="pagination"] > span')
+                    .should('exist')
+                    .and('contain.text', 'Page 1 of 5');
+                }
+            })
+        });
+    });
+    context('Quando navego até a página 5 e valido o botão de "next" desabilitado', () => {
```

Comment:
Em um teste de componente com Cypress, tal cenário poderia ser testado da seguinte forma:

```js
it('renders in the last of two pages (Next button is disabled)', () => {
  const props = {
    currentPage: 2,
    paginationInfo: {
      totalPages: 2,
      limit: 5
    },
    ...defaultProps,
  }

  cy.mount(<Pagination {...props} />)

  cy.contains('button', 'Prev').should('be.enabled').and('be.visible')
  cy.contains('button', 'Next').should('be.disabled').and('be.visible')
  cy.contains('span', 'Page 2 of 2').should('be.visible')
})

```

Mais simples, não acha? Nada de `for` _loops_ e tal.